sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -369,6 +458,9 @@ class RocksDB(
       flushOptions.close()
       dbOptions.close()
       dbLogger.close()
+      synchronized {
+        latestCheckpoint.foreach(_.close())
```

Comment:
do we still need this? looks like the next `silentDeleteRecursively` can delete all local temp directories?

Reply:
I think you are right, the checkpoint don't contain any resource other than the local temp dir which is going to be closed by the below statement anyway. But to be future proof(we may add other resource to it), I prefer to keep the free the resources explicitly here.